sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidget.kt

```diff
@@ -8,24 +8,84 @@ import com.mapbox.maps.MapboxExperimental
  *
  * @param bitmap bitmap used to draw widget
  * @param position position of widget
- * @param marginX horizontal margin in pixels
- * @param marginY vertical margin in pixels
  */
 @MapboxExperimental
 open class BitmapWidget @JvmOverloads constructor(
   bitmap: Bitmap,
-  position: WidgetPosition = WidgetPosition(
-    vertical = WidgetPosition.Vertical.TOP,
-    horizontal = WidgetPosition.Horizontal.LEFT,
-  ),
-  marginX: Float = 0f,
-  marginY: Float = 0f,
+  position: WidgetPosition = WidgetPosition {
+    verticalAlignment = WidgetPosition.Vertical.TOP
+    horizontalAlignment = WidgetPosition.Horizontal.LEFT
+    offsetX = 0f
+    offsetY = 0f
+  }
 ) : Widget() {
+  /**
+   * The deprecated constructor for BitmapWidget.
+   *
+   * @param bitmap bitmap used to draw widget
+   * @param position position of widget
+   * @param marginX horizontal margin in pixels
+   * @param marginY vertical margin in pixels
+   */
+  @Deprecated(
+    message = "Constructor with margins is deprecated, the offset parameters has been merged into " +
+      "the WidgetPosition class, and the legacy constructor might be removed in future releases.",
+    replaceWith = ReplaceWith("BitmapWidget(context, position)")
+  )
+  constructor(
+    bitmap: Bitmap,
+    position: WidgetPosition = WidgetPosition(
```

Comment:
nit: 
maybe name it as default and move to constants

Reply:
This is deprecated constructor, we should limit the scope to avoid confusion and deprecation in multiple places.